src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -192,12 +248,41 @@ private static InputValueDefinitionNode GenerateSimpleInputType(NameNode name, F
                 location: null,
                 field.Name,
                 new StringValueNode($"Input for field {field.Name} on type {inputTypeName}"),
-                type,
+                databaseType is DatabaseType.MSSQL ? type.NullableType() : type,
```

Comment:
Yes. Because for nested insertion, you can skip providing value for all the relationship fields irrespective of whether the relationship is nullable or not.